trihoang0809/tourific
# Update Backend Validation with Zod and Improve Error Handling

# Description


This commit includes updates related to backend validation using Zod, error display, and adjustments in routing and schema for trip creation.

## Issues

N/A





## Screenshots



## Test

- Start the Backend Server: Ensure that your backend server is running and accessible.
- Open Postman and create a new POST request.
- Set the request URL to the endpoint where trips are created (e.g., http://localhost:your_port/trips).
- Set the request body to JSON format with the necessary fields according to the updated schema. For example:

```json
{
  "name": "California Road Trip",
  "startDate": "2024-05-01T00:00:00Z",
  "endDate": "2024-05-07T23:59:59Z",
  "location": {
    "address": "123 Main St",
    "citystate": "Los Angeles, CA",
    "latitude": 34.0522,
    "longitude": -118.2437,
    "radius": 10
  }
}
```

Ensure that the request body adheres to the validation rules specified in the updated schema.


## Possible Downsides

N/A


## Additional Documentations

N/A
